ext/ext/src/lib.rs

```diff
@@ -0,0 +1,141 @@
+// lib.rs
+
+// Copyright (C) 2020-2021 The Nitrocli Developers
+// SPDX-License-Identifier: GPL-3.0-or-later
+
+use std::env;
+use std::ffi;
+use std::process;
+
+use anyhow::Context as _;
+
+#[derive(Debug)]
+pub struct Context {
+  pub nitrocli: ffi::OsString,
+  pub resolved_usb_path: Option<String>,
+  pub verbosity: Option<u8>,
+  pub project_dirs: directories::ProjectDirs,
```

Comment:
What do you think about making these members private? Feels like they are implementation details for the most part. The project directories I'd consider exposing just as paths, so that we don't expose some third party crate's data type across crate boundaries: `Context::cache_dir() -> &Path` (just expose what we need right now).

What do you think?